src/xdocs/writingjavadocchecks.xml.vm

```diff
@@ -558,11 +558,123 @@ JAVADOC -> JAVADOC [0:0]
+        next javadoc. The following example illustrates how to use this property.
+    </p>
+
+    <p>
+    Input:
+    </p>
+
+    <source><![CDATA[
+/**
+  * <body>
+  * <p> This class is only meant for testing. </p>
+  * <p> In html, closing all tags is not necessary.
+  * <li> neither is opening every tag <p> </li>
+  * </body>
+  *
+  * @see "https://www.w3.org/TR/html51/syntax.html#optional-start-and-end-tags"
+  */
+ public class Test {
+     /** <p><p> paraception </p> </p> */
+     private int field1;
+
+     /**<li> paraTags should be opened</p> list isn't nested in parse tree </li>*/
+     private int field2;
+
+     /**
+      * <p> this paragraph is closed and would be nested in javadoc tree </p>
+      * <li> this list has an <p> unclosed para, but still the list would get nested </li>
+      */
+     private int field3;
+
+     /**
+      * <li> Complete <p> nesting </p> </li>
+      * <tr> Zero </p> nesting despite `tr` is closed </tr>
+      */
+     private int field4;
+}
+    ]]></source>
+
+    <p>
+        Output with <tt>violateExecutionOnNonTightHtml</tt> set to false:
+    </p>
+
+    <table>
+      <tr>
+        <td>
+          <source><![CDATA[
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC
+"-//Puppy Crawl//DTD Check Configuration 1.3//EN"
+"http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
+<module name="Checker">
+  <property name="charset" value="UTF-8"/>
+    <module name="TreeWalker">
+      <module name="JavadocParagraph">
+        <property name="violateExecutionOnNonTightHtml" value="false"/>
+      </module>
+    </module>
+</module>
+          ]]></source>
+        </td>
+
+        <td>
+          <source><![CDATA[
+Starting audit...
+[ERROR] Test.java:11: Redundant <p> tag. [JavadocParagraph]
+Audit done.
+Checkstyle ends with 1 errors.
+          ]]></source>
+        </td>
+      </tr>
+    </table>
+
+    <p>
+        Output with <tt>violateExecutionOnNonTightHtml</tt> set to true:
+    </p>
+
+    <table>
+      <tr>
+        <td>
+          <source><![CDATA[
+<?xml version="1.0"?>
+<!DOCTYPE module PUBLIC
+"-//Puppy Crawl//DTD Check Configuration 1.3//EN"
+"http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
+<module name="Checker">
+<property name="charset" value="UTF-8"/>
+<module name="TreeWalker">
+<module name="JavadocParagraph">
+<property name="violateExecutionOnNonTightHtml" value="true"/>
+</module>
+</module>
+</module>
+          ]]></source>
+        </td>
+
+        <td>
+          <source><![CDATA[
+Starting audit...
+[ERROR] Test.java:4: Unclosed HTML tag found: p [JavadocParagraph]
+[ERROR] Test.java:11: Redundant <p> tag. [JavadocParagraph]
+[ERROR] Test.java:11: Unclosed HTML tag found: p [JavadocParagraph]
+[ERROR] Test.java:14: Unclosed HTML tag found: li [JavadocParagraph]
+[ERROR] Test.java:19: Unclosed HTML tag found: p [JavadocParagraph]
+[ERROR] Test.java:25: Unclosed HTML tag found: tr [JavadocParagraph]
```

Comment:
why we complain about TR ?

Reply:
> `<tr> Zero </p> nesting despite `tr` is closed </tr>`

`</p>` breaks apart HTML_ELEMENT into a separate open and tag close.
This is similar to another issue, but I don't remember the number.

Reply:
Found the issue. https://github.com/checkstyle/checkstyle/issues/3570

Reply:
I think it's because the production for rule `tr` in grammar and all other tags for that matter only have `...TagStart` rules and no `...TagEnds`.